### PR TITLE
Magiclysm Attunements: add frozen winds, auroral wave, and immolate spells.

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

SUMMARY: Mods "Adds three new Attunement spells (frozen winds, auroral wave, immolate)"

#### Purpose of change
Part of [Magiclysm Attunements project](https://github.com/CleverRaven/Cataclysm-DDA/issues/45011).

#### Describe the solution
Adds the Frozen Winds spell, which does medium but significant cold damage and a firm push effect in a very wide line AOE

Adds the Boreal mage Auroral Wave spell, which has a focused area of cold and bashing damage and a wider cone of just cold damage.

Adds the Cleansing flame Immolate spell, which just destroys everything in a pretty wide AOE.

#### Describe alternatives you've considered
I don't know magiclysm balance too well. Based these on existing spells but at some point gathering from the disparity I'm seeing, a global audit of magiclysm balance for spells may be in order.

#### Testing
Pending

### CODE REVIEW

File: data/mods/Magiclysm/Spells/attunements/Boreal_Mage.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    "min_damage": 30,
+    "max_damage": 100,
+    "damage_increment": 2.0,
+    "min_range": 4,
+    "max_range": 12,
+    "range_increment": 0.25,
+    "min_aoe": 45,
+    "max_aoe": 120,
+    "aoe_increment": 3,
+    "spell_class": "GLACIER_MAGE",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/attunements/Boreal_Mage.json:48]:
```suggestion
```

File: data/mods/Magiclysm/Spells/attunements/Glacier_Mage.json [hunk 2]
```diff
@@ -47,5 +47,54 @@
...
+    "sound_description": "a whoosh",
+    "damage_type": "cold"
+  },
+  {
+    "id": "frozen_winds_push",
+    "type": "SPELL",
+    "name": "Frozen Winds push effect",
+    "description": "This is the push effect portion of the Frozen Winds spell.",
+    "effect": "directed_push",
+    "effect": "attack",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/attunements/Glacier_Mage.json:85]:
```suggestion
```

File: data/mods/Magiclysm/Spells/attunements/Cleansing_Flame.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    "sound_description": "a searing whoosh",
+    "energy_source": "MANA"
+  },
+  {
+    "id": "immolate_push",
+    "type": "SPELL",
+    "name": "Immolate push effect",
+    "description": "The immolate spell is powerful enough to push targets from its epicenter.",
+    "effect": "directed_push",
+    "effect": "attack",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/attunements/Cleansing_Flame.json:53]:
```suggestion
```

**Comment by KorGgenT** [comment]
image
there needs to be an actual spell for this, in addition to the ter_furn_transform you have here.

**Review by KorGgenT — APPROVED** [review]
OK i'm going to sign off on this. i'd like it to be merged before stable as it's attunements and i'd like as much of those done as humanly possible before string freeze. I've done enough edits to this PR that i should probably not be the one to merge it.

**Comment by actual-nh** [comment]
@KorGgenT: At some point - after 0.F, I would think! - could we discuss appropriate material for the [wiki](http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Category:Magiclysm) relating to attunements? I've been working on "regular" spells so far. Thanks!

**Comment by KorGgenT** [comment]
Yeah absolutely

**Pull request merged**